aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,486 @@
+
+The following table shows `data annotations <https://msdn.microsoft.com/en-us/library/system.componentmodel.dataannotations.aspx>`__ attributes that the input tag helper will map to specific input types:
+
++-------------------------------+--------------------+
+|Attribute                      |  Input Type        |  
++===============================+====================+
+|[EmailAddress]                 |  type="email"      |
++-------------------------------+--------------------+  
+|[Url]                          |  type="url"        |
++-------------------------------+--------------------+  
+|[HiddenInput]                  |  type="hidden"     |
++-------------------------------+--------------------+  
+|Phone]                         |  type="tel"        |
++-------------------------------+--------------------+   
+|[DataType(DataType.Password)]  |  type="password"   |
++-------------------------------+--------------------+  
+|[DataType(DataType.Date)]      |  type="date"       |
++-------------------------------+--------------------+  
+|[DataType(DataType.Time)]      |  type="time"       |
++-------------------------------+--------------------+  
+ 
+Sample: 
+ 
+.. literalinclude::  forms/sample/final/ViewModels/RegisterViewModel.cs
+  :language: c#
+
+.. literalinclude::  forms/sample/final/Views/Demo/RegisterInput.cshtml
+  :language: HTML
+
+The code above generates the following HTML:
+
+.. code-block:: HTML
+
+    <form method="post" action="/Demo/RegisterInput">
+      Email:  
+      <input type="email" data-val="true" 
+             data-val-email="The Email Address field is not a valid e-mail address." 
+             data-val-required="The Email Address field is required." 
+             id="Email" name="Email" value="" /> <br>
+      Password: 
+      <input type="password" data-val="true" 
+             data-val-required="The Password field is required." 
+             id="Password" name="Password" /><br>
+      <button type="submit">Register</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
+
+The data annotations applied to the ``Email`` and ``Password`` properties generated meta data on the model in the form of `HTML5 <https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5>`__ ``data-val-*`` attributes (see :doc:`/mvc/models/validation`). (``data-val-`` specifies data validation.) The Input Tag Helper consumes the model metadata and produces HTML5-compatible attributes describing the validators to attach to the input fields. This provides unobtrusive HTML5 and `jQuery <https://jquery.com/>`__ validation. The unobtrusive attributes have the format ``data-val-rule="Error Message"``, where rule is the validation rule (such as required, email format, minimum string length, and maximum string length). If an error message is provided in the attribute, it is displayed as the value for the ``data-val-rule`` attribute. 
```

Comment:
The validation attributes do none of the validation mentioned in the first sentence. The later sentences are closer but keep in mind that annotations are themselves metadata (one word) and .NET and HTML attributes are very different beasts.